controllers/users.js

```diff
@@ -738,11 +738,34 @@ const filterUsers = async (req, res) => {
       return res.boom.badRequest("filter for item not provided");
     }
 
-    const users = await dataAccess.retreiveFilteredUsers(req.query);
+    const { page, limit } = req.query;
+    const pageNumber = parseInt(page) || 0;
+    const limitNumber = parseInt(limit) || 100;
+    const skip = (pageNumber - 1) * limitNumber;
+
+    const users = await dataAccess.retreiveFilteredUsers(req.query, skip, limitNumber);
+    const totalCount = await users.length;
+    const totalPages = Math.ceil(totalCount / limitNumber);
+    const nextPage = pageNumber < totalPages - 1 ? pageNumber + 1 : null;
+    const prevPage = pageNumber > 0 ? pageNumber - 1 : null;
+
+    const paginationLinks = {
+      next: nextPage ? getFilteredPaginationLink(req.query, nextPage, limitNumber) : null,
+      prev: prevPage ? getFilteredPaginationLink(req.query, prevPage, limitNumber) : null,
+    };
+
```

Comment:
- can we create a function for this which returns us the required value for the pagination object below, this way the controller does not grow big?

Reply:
created the function which will be returning the pagination object